### ISSUE TITLE
crates under 3/l downloaded incorrectly

### ISSUE THREAD

**Issue by Kepler-Br** [panamax-rs/panamax#83]
```shell

% cargo install cargo-cache
    Updating `panamax` index
  Installing cargo-cache v0.8.3
error: failed to compile `cargo-cache v0.8.3`, intermediate artifacts can be found at `/tmp/cargo-installGZoqCc`

Caused by:
  failed to download from `http://crates.dw/3/l/log/0.4.17/log-0.4.17.crate`

Caused by:
  failed to get 200 response from `http://crates.dw/3/l/log/0.4.17/log-0.4.17.crate`, got 404
```

I've checked index and there is an index file under `/3/l/log`, but crates folder contains log crate under path `3/log` which is not correct

I've tried with dockerized version of latest commit on master

**Comment by jbg** [comment]
You can't serve the files directly without some rewrites, see the sample nginx config in the repo: https://github.com/panamax-rs/panamax/blob/master/nginx.sample.conf

I'm not sure why the decision was made to have the on-disk layout differ from the served layout. I wanted to use panamax to update a mirror which is mounted into build tasks (so the crates base URL is a `file://` URL) but it doesn't work because of this same issue.

**Comment by k3d3** [comment]
Oh shoot, I didn't realize the formats differed - the intention was to have them be the same format. I could've sworn the index used to be 3/<crate> rather than 3/<1st>/<crate>.

Ill try to get this fixed - sorry about that.

**Comment by jbg** [comment]
Actually even with the rewrites in place, there seem to be issues with crates that have short names.

```
[16/Jan/2023:05:40:15 +0000] "GET /crates/3/n/num/0.3.1/num-0.3.1.crate HTTP/1.1" 404 153 "-" "cargo 1.66.1 (ad779e08b 2023-01-10)"
```

**Comment by jbg** [comment]
I replaced the rewrites with this and it seems to work:

```
rewrite "^/crates/3/[^/]/([^/]+)/([^/]+)/([^/]+)$" "/crates/3/$1/$2/$3";
```

**Comment by k3d3** [comment]
This should be fixed as of 1.0.8. Feel free to reopen this issue if you're still having trouble.
